ocode_python/core/semantic_context.py

```
@@ -0,0 +1,779 @@
+        self.embedding_cache = EmbeddingCache(self.cache_dir)
+
+        # Initialize embedding model if available
+        self.embeddings_model = None
+        self.model_version = "fallback"
+
+        if EMBEDDINGS_AVAILABLE:
+            try:
+                self.embeddings_model = SentenceTransformer("all-MiniLM-L6-v2")
+                self.model_version = "all-MiniLM-L6-v2"
+                logging.info("Semantic embeddings enabled with sentence-transformers")
+            except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)